core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -107,6 +109,11 @@ public static Environment prepareEnvironment(Settings input, Terminal terminal,
             throw new SettingsException("multiple settings files found with suffixes: "
                 + Strings.collectionToDelimitedString(foundSuffixes, ","));
         }
+        if (foundSuffixes.contains(".yml") || foundSuffixes.contains(".json")) {
+            final DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(InternalSettingsPreparer.class));
```

Comment:
Sadly, the logging here happens before we've fully configured logging. This means that it will appear in the console log with a default format, and before we've configured deprecation logging so it won't end up in the deprecation log on disk.